src/value.rs

```diff
@@ -136,33 +136,50 @@ impl Value {
 }
 
 /// A request contains multiple entries to be written into LSM tree.
+#[derive(Clone)]
 pub struct Request {
+    /// Entries contained in this request
     pub entries: Vec<Entry>,
+    /// Offset in vLog (will be updated upon processing the request)
+    pub ptrs: Vec<ValuePointer>,
+    /// Use channel to notify that the value has been persisted to disk
+    pub done: Option<Sender<Result<()>>>,
 }
 
 /// `ValuePointer` records the position of value saved in value log.
+#[derive(Clone, Default, Debug)]
 pub struct ValuePointer {
     pub file_id: u32,
     pub len: u32,
     pub offset: u32,
 }
 
-/// `EntryReader` reads entries from `BufReader`.
+impl ValuePointer {
+    pub fn decode(&mut self, mut bytes: &[u8]) {
+        self.file_id = bytes.get_u32();
+        self.len = bytes.get_u32();
+        self.offset = bytes.get_u32();
+    }
+
+    pub fn encode(&self, buf: &mut BytesMut) {
+        buf.put_u32(self.file_id);
```

Comment:
How about using little endian? Given that most of processors use little endians, so it will be a no-op.

Reply:
Now we use little-endian for vptr encoding.

Previously in SSTs, we used big-endian to encode length. We may later determine what endian to use in all places.